src/items/item.cpp

```diff
@@ -108,6 +108,35 @@ void Item::setImbuement(uint8_t slot, uint16_t id, int32_t duration)
 	setCustomAttribute(key, value);
 }
 
+void Item::addImbuement(uint8_t slot, uint16_t id, int32_t duration)
+{
```

Comment:
Really nice code, Ill put tbis inside set imbuement though and kill the add imbuement function.